### PR TITLE
fix(tabletoolbarsearch): unique id passed to search should be a string

### LINKED ISSUE #4153: TableToolbarSearch - Warning: Failed prop type: Invalid prop `id` of type `number` supplied to `Search`, expected `string`.
## What package(s) are you using?

- [x] `carbon-components`
- [x] `carbon-components-react`

## Detailed description

> Describe in detail the issue you're having.

`Search` is not provided an `id` prop of type `string` from `TableToolbarSearch`. It is provided a number.

> Is this issue related to a specific component?

`TableToolbarSearch`

> What did you expect to happen? What happened instead? What would you like to
> see changed?

`setupGetInstanceId()` returns a number, which is used for the `id` prop on `Search`.

https://github.com/carbon-design-system/carbon/blob/1a435757c9f5b69d6f21f9a5269bb8ba0c24747c/packages/react/src/tools/setupGetInstanceId.js#L15

https://github.com/carbon-design-system/carbon/blob/b2636f786592dfbfbee2bf05f55c40719db15f0b/packages/react/src/components/DataTable/TableToolbarSearch.js#L45

https://github.com/carbon-design-system/carbon/blob/b2636f786592dfbfbee2bf05f55c40719db15f0b/packages/react/src/components/DataTable/TableToolbarSearch.js#L93

This should be converted to a `string`, to match proptypes for `Search`

This overall manifests as a proptype warning in our project:
```
Warning: Failed prop type: Invalid prop `id` of type `number` supplied to `Search`, expected `string`.
```

> What browser are you working in?

Chrome

> What version of the Carbon Design System are you using?

```
"carbon-components": "^10.6.4",
"carbon-components-react": "^7.6.4",
```

> What offering/product do you work on? Any pressing ship or release dates we
> should be aware of?

Watson IoT PAL - It causes our storybook snapshot tests to fail. Was noticed while upgrading from 10.4

### PR DESCRIPTION
**Pull request by tay1orjones**
Closes #4153 

Modify `id` prop to be a string when no id is provided, to pass prop type validation

#### Changelog

**Changed**

- If no `id` is passed to TableToolbarSearch, a unique id is provided in string format

#### Testing / Reviewing

Check out the storybook - react dev tools should show the prop no longer as a number, but as a string.

I added a passing test with no `id` configured. It should fail on console error (proptype warning) without the aforementioned change.

### CODE REVIEW

File: packages/react/src/components/DataTable/__tests__/TableToolbarSearch-test.js [hunk 1]
```diff
@@ -20,4 +20,10 @@ describe('DataTable.TableToolbarSearch', () => {
     );
     expect(wrapper).toMatchSnapshot();
   });
+  it('should render without specifying an id', () => {
+    const wrapper = mount(
+      <TableToolbarSearch className="custom-class" onChange={jest.fn()} />
+    );
+    expect(wrapper).toMatchSnapshot();
+  });
```
Comment (joshblack) [inline packages/react/src/components/DataTable/__tests__/TableToolbarSearch-test.js]: What are we looking to test in this bit? Just that no errors or warnings are logged?

**Review by emyarod — APPROVED** [review]
looks good to me!

File: packages/react/src/components/DataTable/__tests__/TableToolbarSearch-test.js (hunk 1 above)
Comment (tay1orjones) [reply packages/react/src/components/DataTable/__tests__/TableToolbarSearch-test.js]: Yeah, let me update that to a more explicit test against the `id` prop value. Will push momentarily.

**Review by jnm2377 — APPROVED** [review]
LGTM! 🎉 thanks for the contribution! Will merge this after our code freeze!

**Review by joshblack — APPROVED** [review]

**Pull request merged**